docs/content/docs/recipes/audience-react.md

```diff
@@ -0,0 +1,396 @@
+
+{{< callout note >}}
+
+This tutorial assumes that you are familiar with the [Fluid Framework Overview]({{< relref "/docs/_index.md" >}}) and that you have completed the [QuickStart]({{< relref "quick-start.md" >}}). You should also be familiar with the basics of [React](https://reactjs.org/), [creating React projects](https://reactjs.org/docs/create-a-new-react-app.html#create-react-app), and [React Hooks](https://reactjs.org/docs/hooks-intro.html).
+
+{{< /callout >}}
+
+## Create the project
+
+1. Open a Command Prompt and navigate to the parent folder where you want to create the project; e.g., `c:\My Fluid Projects`.
+1. Run the following command at the prompt. (Note that the CLI is np**x**, not npm. It was installed when you installed Node.js.)
+
+    ```dotnetcli
+    npx create-react-app fluid-audience-tutorial
+    ```
+
+1. The project is created in a subfolder named `fluid-audience-tutorial`. Navigate to it with the command `cd fluid-audience-tutorial`.
+
+1. The project uses the following Fluid libraries:
+
+    |Library |Description |
+    |---|---|
+    | `fluid-framework`    |Contains the SharedMap [distributed data structure]({{< relref "dds.md" >}}) that synchronizes data across clients.|
+    | `@fluidframework/azure-client`   |Defines the connection to a Fluid service server and defines the starting schema for the [Fluid container][].|
+    | `@fluidframework/test-client-utils`   |Defines the `InsecureTokenProvider` needed to create the connection to a Fluid Service.|
+    {.table}
+
+    Run the following command to install the libraries.
+
+    ```dotnetcli
+    npm install @fluidframework/azure-client @fluidframework/test-client-utils fluid-framework
+    ```
+
+## Code the project
+
+### Set up state variables and component view
+
+1. Open the file `\src\App.js` in the code editor. Delete all the default `import` statements. Then delete all the markup from the `return` statement. Then add import statements for components and React hooks. Note that we will be implementing the components in the later steps. The file should look like the following:
+
+```js
+  import { useState, useCallback } from "react";
+  import { AudienceDisplay } from "./AudienceDisplay";
+  import { UserIdSelection } from "./UserIdSelection"
+
+  export const App = () => {
+    // TODO 1: Define state variables to handle view changes and user input
+    return (
+    // TODO 2: Return view components
+    );
+  }
+```
+
+1. Replace `TODO 1` with the following code. Note that the values `userId` and `containerId` will come from a `UserIdSelection` component through the `handleSelectUser` function. The `displayAudience` variable is used to switch between the member list view and the userId selection view. In the case that a user inputs an invalid `audienceId` they will be redirected to the userId selection view by the `handleContainerNotFound` function.
```

Comment:
Focus the content around why this code is here and not just what is happening. E.g. What does `handleSelectUser` do and why are we setting all these properties?

Avoid referring to concepts that have not been introduced yet. When you do need to do this make sure you tell the reader where they will be introduced. Otherwise people feel like they missed something. E.g. you talk about the member list view and the userId selection view but those are in the next code block.

Use class names (with optional descriptors) instead of just descriptors. Eg. `AudienceDisplay` vs "member list view". 

```suggestion
1. Replace `TODO 1` with the following code. This code initializes local state variables that will be used within the application. `displayAudience` determines if we render the `AudienceDisplay` component or the `UserIdSelection` component (see `TODO 2`). `userId` contains the user identifier to connect to the container with and `containerId` is the container to load. `handleSelectUser` and `handleContainerNotFound` are passed in as callbacks to the two views and manage state transitions. `handleSelectUser` is called when attempting to create/load a container. `handleContainerNotFound` is called when creating/loading a container fails.

Note, the values `userId` and `containerId` will come from a `UserIdSelection` component through the `handleSelectUser` function. 
```